pytest_pretty/__init__.py

```diff
@@ -98,3 +100,48 @@ def pytest_configure(config):
     custom_reporter = CustomTerminalReporter(config, sys.stdout)
     config.pluginmanager.unregister(standard_reporter)
     config.pluginmanager.register(custom_reporter, 'terminalreporter')
+
+
+ansi_escape = re.compile(r'(?:\x1B[@-_]|[\x80-\x9F])[0-?]*[ -/]*[@-~]')
+stat_re = re.compile(r'(\d+) (\w+)')
+
+
+def _new_outcomes(obj):
+    def parseoutcomes():
+        lines_with_stats = dropwhile(lambda x: 'Results' not in x, obj.outlines)
+        next(lines_with_stats)  # drop Results line
+        res = {}
+        for i, line in enumerate(lines_with_stats):
+            line = ansi_escape.sub('', line).strip()  # clean colors
+            match = stat_re.match(line)
+
+            if match is None:
+                break
+
+            res[match.group(2)] = int(match.group(1))
+
+        return res
+
+    return parseoutcomes
+
+
+class PytesterWrapper:
+    def __init__(self, pytester):
+        object.__setattr__(self, 'pytester', pytester)
```

Comment:
why not 
```suggestion
        self.pytester = pytester
```
?

Reply:
or even `_pytester` since it's only used internally.

Reply:
Because of the infinite recursion problem:
```
/workspace/pytest-pretty/pytest_pretty/__init__.py:138: in __init__
    self._pytester = pytester
/workspace/pytest-pretty/pytest_pretty/__init__.py:150: in __setattr__
    setattr(self._pytester, name, value)
/workspace/pytest-pretty/pytest_pretty/__init__.py:147: in __getattr__
    return getattr(self._pytester, name)
/workspace/pytest-pretty/pytest_pretty/__init__.py:147: in __getattr__
    return getattr(self._pytester, name)
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
Results (0.20s):
```

https://python-reference.readthedocs.io/en/latest/docs/dunderattr/setattr.html#remarks